compiler/ilgen/MethodBuilder.cpp

```diff
@@ -80,16 +86,22 @@ namespace OMR
 
 MethodBuilder::MethodBuilder(TR::TypeDictionary *types, OMR::VirtualMachineState *vmState)
    : TR::IlBuilder(asMethodBuilder(), types),
+   // Note: Any allocations in the following memory region that will get deallocated by C++ destruction semantics along with this MethodBuilder class, should instead
+   // be explicitly cleared / deallocated in the MethodBuilder destructor *before* the region itself is destroyed in order to avoid undefined behaviour from invalid
+   // memory references.
```

Comment:
Sorry. My comment wasn't well constructed, mostly because I think I reviewed the file before all the changes had been made to it. I'll accept the comment being at the constructor.